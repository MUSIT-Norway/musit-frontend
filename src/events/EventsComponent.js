import React from 'react';
import { hashHistory } from 'react-router';

import ObservationControlGrid from './EventsGrid';
import ObservationControlComponent from './EventsSideBar';

import Layout from '../layout';
import Breadcrumb from '../layout/Breadcrumb';
import Toolbar from '../layout/Toolbar';

export default class ObservationControlGridShow extends React.Component {
  static propTypes = {
    observationControlGridData: React.PropTypes.arrayOf(React.PropTypes.object),
    params: React.PropTypes.object,
    route: React.PropTypes.object,
    loadControlAndObservations: React.PropTypes.func.isRequired,
    loadStorageObj: React.PropTypes.func.isRequired,
    path: React.PropTypes.arrayOf(React.PropTypes.object)
  }

  constructor(props) {
    super(props);
    this.props.params.id = this.props.params.id * 1;
    this.state = {
      showObservations: this.props.route.showObservations,
      showControls: this.props.route.showControls
    };
  }

  componentWillMount() {
    this.props.loadControlAndObservations(this.props.params.id, this.props.user.museumId);
    this.props.loadStorageObj(this.props.params.id, this.props.user.museumId);
  }

  makeToolbar() {
    return <Toolbar
      showRight={this.state.showControls}
      showLeft={this.state.showObservations}
      labelRight="Kontroller"
      labelLeft="Observasjoner"
      placeHolderSearch="Filtrer i liste"
      clickShowRight={() => this.setState({ ...this.state, showControls: !this.state.showControls })}
      clickShowLeft={() => this.setState({ ...this.state, showObservations: !this.state.showObservations })}
    />;
  }

  makeLeftMenu() {
    return <div style={{ paddingTop: 10 }}>
      <ObservationControlComponent
        id={this.props.params.id}
        selectObservation
        selectControl
        onClickNewObservation={() => hashHistory.push(`/magasin/${this.props.params.id}/observation/add`)}
        onClickNewControl={() => hashHistory.push(`/magasin/${this.props.params.id}/control/add`)}
      />
    </div>;
  }

  makeContent() {
    return <ObservationControlGrid
      id={this.props.params.id}
      tableData={this.props.observationControlGridData.filter((e) => {
        if (e.eventType && this.state.showControls && this.state.showObservations) {
          return true;
        } else if (e.eventType && this.state.showControls) {
          return e.eventType.toLowerCase() === 'control';
        } else if (e.eventType && this.state.showObservations) {
          return e.eventType.toLowerCase() === 'observation';
        }
        return false;
      })}
    />;
  }

  render() {
    return (
      <Layout
        title="Magasin"
        breadcrumb={
          <Breadcrumb
            node={this.props.rootNode}
            onClickCrumb={(node) => hashHistory.push(node.url)}
            allActive
          />
        }
        toolbar={this.makeToolbar()}
        leftMenu={this.makeLeftMenu()}
        content={this.makeContent()}
      />
    );
  }
}