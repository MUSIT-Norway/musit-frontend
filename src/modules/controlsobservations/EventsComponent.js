import React from 'react';
import EventsGrid from './EventsGrid';
import EventsLeftMenu from './EventsLeftMenu';
import Layout from '../../components/layout';
import Breadcrumb from '../../components/layout/Breadcrumb';
import Toolbar from '../../components/layout/Toolbar';
import { hashHistory } from 'react-router';
import { I18n } from 'react-i18nify';
import inject from 'react-rxjs/dist/RxInject';
import store$, { clearEvents$, loadRootNode$, loadEvents$ } from './eventsStore';
import Loader from 'react-loader';
import Config from '../../config';

export class EventsComponent extends React.Component {
  static propTypes = {
    appSession: React.PropTypes.object,
    store: React.PropTypes.object,
    params: React.PropTypes.object,
    route: React.PropTypes.object,
    loadEvents: React.PropTypes.func.isRequired,
    loadRootNode: React.PropTypes.func.isRequired,
    clearEvents: React.PropTypes.func.isRequired,
    loader: React.PropTypes.element.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      showObservations: this.props.route.showObservations,
      showControls: this.props.route.showControls
    };
  }

  static defaultProps = {
    loader: <Loader loaded={false} />
  };

  componentWillMount() {
    this.props.clearEvents();
    this.props.loadEvents({
      nodeId: this.props.params.id,
      museumId: this.props.appSession.getMuseumId(),
      token: this.props.appSession.getAccessToken()
    });
    this.props.loadRootNode({
      id: this.props.params.id,
      museumId: this.props.appSession.getMuseumId(),
      token: this.props.appSession.getAccessToken()
    });
  }

  makeToolbar() {
    return (
      <Toolbar
        showRight={this.state.showControls}
        showLeft={this.state.showObservations}
        labelRight={I18n.t('musit.grid.button.controls')}
        labelLeft={I18n.t('musit.grid.button.observations')}
        placeHolderSearch={I18n.t('musit.grid.search.placeHolder')}
        clickShowRight={() =>
          this.setState({ ...this.state, showControls: !this.state.showControls })}
        clickShowLeft={() =>
          this.setState({
            ...this.state,
            showObservations: !this.state.showObservations
          })}
      />
    );
  }

  makeLeftMenu() {
    const nodeId = this.props.params.id;
    const appSession = this.props.appSession;
    return (
      <div style={{ paddingTop: 10 }}>
        <EventsLeftMenu
          id={this.props.params.id}
          selectObservation
          selectControl
          onClickNewObservation={() =>
            hashHistory.push(
              Config.magasin.urls.client.storagefacility.addObservation(
                nodeId,
                appSession
              )
            )}
          onClickNewControl={() =>
            hashHistory.push(
              Config.magasin.urls.client.storagefacility.addControl(nodeId, appSession)
            )}
        />
      </div>
    );
  }

  makeContent() {
    if (this.props.store.loading) {
      return this.props.loader;
    }
    const filtered = this.props.store.data.filter(e => {
      if (e.eventType && this.state.showControls && this.state.showObservations) {
        return true;
      } else if (e.eventType && this.state.showControls) {
        return e.eventType.toLowerCase() === 'control';
      } else if (e.eventType && this.state.showObservations) {
        return e.eventType.toLowerCase() === 'observation';
      }
      return false;
    });
    if (filtered.length === 0) {
      return (
        <div style={{ textAlign: 'center', color: 'grey' }}>
          {I18n.t('musit.events.noData')}
        </div>
      );
    }
    const nodeId = this.props.params.id;
    const appSession = this.props.appSession;
    return (
      <EventsGrid
        id={nodeId}
        showControl={ctl =>
          hashHistory.push(
            Config.magasin.urls.client.storagefacility.viewControl(
              nodeId,
              ctl.id,
              appSession
            )
          )}
        showObservation={obs =>
          hashHistory.push(
            Config.magasin.urls.client.storagefacility.viewObservation(
              nodeId,
              obs.id,
              appSession
            )
          )}
        tableData={filtered}
      />
    );
  }

  showNodes(node) {
    const appSession = this.props.appSession;
    if (node && node.id) {
      hashHistory.push(
        Config.magasin.urls.client.storagefacility.goToNode(node.id, appSession)
      );
    } else {
      hashHistory.push(Config.magasin.urls.client.storagefacility.goToRoot(appSession));
    }
  }

  render() {
    return (
      <Layout
        title={I18n.t('musit.storageUnits.title')}
        breadcrumb={
          <Breadcrumb
            node={this.props.store.rootNode}
            onClickCrumb={node => this.showNodes(node)}
            allActive
          />
        }
        toolbar={this.makeToolbar()}
        leftMenu={this.makeLeftMenu()}
        content={this.makeContent()}
      />
    );
  }
}

const data = {
  appSession$: { type: React.PropTypes.object.isRequired },
  store$
};

const commands = {
  clearEvents$,
  loadRootNode$,
  loadEvents$
};

export default inject(data, commands)(EventsComponent);
