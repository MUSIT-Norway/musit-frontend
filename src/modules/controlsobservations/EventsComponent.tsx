import * as React from 'react';
import * as PropTypes from 'prop-types';
import EventsGrid from './EventsGrid';
import EventsLeftMenu from './EventsLeftMenu';
import Layout from '../../components/layout';
import Breadcrumb from '../../components/layout/Breadcrumb';
import Toolbar from '../../components/layout/Toolbar';
import { I18n } from 'react-i18nify';
import { RxInjectLegacy as inject } from '../../shared/react-rxjs-patch';
import store$, { clearEvents$, loadRootNode$, loadEvents$ } from './eventsStore';
import * as Loader from 'react-loader';
import Config from '../../config';
import { AppSession } from '../../types/appSession';
import { TODO } from '../../types/common';
import { Match } from '../../types/Routes';
import { History } from 'history';

interface EventsComponentProps {
  appSession: AppSession;
  store: TODO;
  match: Match<TODO>;
  route: object;
  loadEvents: Function;
  loadRootNode: Function;
  clearEvents: Function;
  loader: JSX.Element;
  history: History;

  // ---
  showControls: boolean;
  showObservations: boolean;
}

/* Old:
  static propTypes = {
    appSession: PropTypes.object,
    store: PropTypes.object,
    match: PropTypes.object,
    route: PropTypes.object,
    loadEvents: PropTypes.func.isRequired,
    loadRootNode: PropTypes.func.isRequired,
    clearEvents: PropTypes.func.isRequired,
    loader: PropTypes.element.isRequired,
    history: PropTypes.object
  };
*/

interface EventsComponentState {
  showControls: boolean;
  showObservations: boolean;
}

export class EventsComponent extends React.Component<
  EventsComponentProps,
  EventsComponentState
> {
  constructor(props: EventsComponentProps) {
    super(props);
    this.state = {
      showObservations: this.props.showObservations,
      showControls: this.props.showControls
    };
  }

  static defaultProps = {
    loader: <Loader loaded={false} />
  };

  componentWillMount() {
    const params = this.props.match.params || {};
    this.props.clearEvents();
    this.props.loadEvents({
      nodeId: params.id,
      museumId: this.props.appSession.museumId,
      token: this.props.appSession.accessToken
    });
    this.props.loadRootNode({
      id: params.id,
      museumId: this.props.appSession.museumId,
      token: this.props.appSession.accessToken
    });
  }

  makeToolbar() {
    return (
      <Toolbar
        showCenter={this.state.showControls}
        showLeft={this.state.showObservations}
        hideRight={true}
        labelCenter={I18n.t('musit.grid.button.controls')}
        labelLeft={I18n.t('musit.grid.button.observations')}
        placeHolderSearch={I18n.t('musit.grid.search.placeHolder')}
        clickShowCenter={() =>
          this.setState(ps => ({
            ...ps,
            showControls: !this.state.showControls
          }))
        }
        clickShowLeft={() =>
          this.setState(ps => ({
            ...ps,
            showObservations: !this.state.showObservations
          }))
        }
      />
    );
  }

  makeLeftMenu(historyPush: Function) {
    const nodeId = this.props.match.params.id;
    const appSession = this.props.appSession;
    return (
      <div style={{ paddingTop: 10 }}>
        {appSession.rolesForModules.storageFacilityWrite && (
          <EventsLeftMenu
            id={nodeId}
            //??? selectObservation
            //??? selectControl
            onClickNewObservation={() =>
              historyPush(
                Config.magasin.urls.client.storagefacility.addObservation(
                  nodeId,
                  appSession
                )
              )
            }
            onClickNewControl={() =>
              historyPush(
                Config.magasin.urls.client.storagefacility.addControl(nodeId, appSession)
              )
            }
          />
        )}
      </div>
    );
  }

  makeContent(historyPush: TODO) {
    if (this.props.store.loading) {
      return this.props.loader;
    }
    const filtered = this.props.store.data.filter((e: TODO) => {
      if (e.eventType && this.state.showControls && this.state.showObservations) {
        return true;
      } else if (e.eventType && this.state.showControls) {
        return e.eventType.toLowerCase() === 'control';
      } else if (e.eventType && this.state.showObservations) {
        return e.eventType.toLowerCase() === 'observation';
      }
      return false;
    });
    if (filtered.length === 0) {
      return (
        <div style={{ textAlign: 'center', color: 'grey' }}>
          {I18n.t('musit.events.noData')}
        </div>
      );
    }
    const nodeId = this.props.match.params.id;
    const appSession = this.props.appSession;
    return (
      <EventsGrid
        id={nodeId}
        showControl={(ctl: TODO) =>
          historyPush(
            Config.magasin.urls.client.storagefacility.viewControl(
              nodeId,
              ctl.id,
              appSession
            )
          )
        }
        showObservation={(obs: TODO) =>
          historyPush(
            Config.magasin.urls.client.storagefacility.viewObservation(
              nodeId,
              obs.id,
              appSession
            )
          )
        }
        tableData={filtered}
      />
    );
  }

  showNodes(node: TODO, historyPush: TODO) {
    const appSession = this.props.appSession;
    if (node && node.nodeId) {
      historyPush(
        Config.magasin.urls.client.storagefacility.goToNode(node.nodeId, appSession)
      );
    } else {
      historyPush(Config.magasin.urls.client.storagefacility.goToRoot(appSession));
    }
  }

  render() {
    return (
      <Layout
        title={I18n.t('musit.storageUnits.title')}
        breadcrumb={
          <Breadcrumb
            node={this.props.store.rootNode}
            onClickCrumb={(node: TODO) => this.showNodes(node, this.props.history.push)}
            allActive
          />
        }
        toolbar={this.makeToolbar()}
        leftMenu={this.makeLeftMenu(this.props.history.push)}
        content={this.makeContent(this.props.history.push)}
      />
    );
  }
}

const data = {
  appSession$: { type: PropTypes.object.isRequired },
  store$
};

const commands = {
  clearEvents$,
  loadRootNode$,
  loadEvents$
};

export default inject(data, commands)(EventsComponent);
