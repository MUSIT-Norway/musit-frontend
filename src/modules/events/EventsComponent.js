import React from 'react';
const Table = require('reactable').Table;
const Tr = require('reactable').Tr;
const Td = require('reactable').Td;
import './eventsComponent.css';

export const EventsComponent = (props) => (
  <div>
    <br />
    <Table
      className="table"
      columns={[
        { key: 'eventDate', label: 'Dato'},
        { key: 'type', label: 'Type hendelse'},
        { key: 'registeredBy', label: 'Utført av' },
        { key: 'note', label: 'Kommentar' },
        { key: 'id', label: '' }
      ]}
      sortable={['type', 'eventDate', 'registeredBy', 'note' ]}
      defaultSort={{ column: 'eventDate', direction: 'desc' }}
      noDataText="Ingen hendelser funnet på objektet"
    >
      {props.events && props.events.map((event, i) =>
        <Tr key={i}>
          <Td column="eventDate">{event.eventDate}</Td>
          <Td column="type">{event.type}</Td>
          <Td column="registeredBy">{event.registeredBy}</Td>
          <Td column="note">{event.note || ''}</Td>
          <Td column="id"><a href="#" onClick={(e) => e.preventDefault()}>Les mer</a></Td>
        </Tr>
      )}
    </Table>
  </div>
);

export default EventsComponent;