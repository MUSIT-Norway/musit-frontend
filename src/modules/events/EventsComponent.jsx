import React from 'react';
import { Table, Tr, Td } from 'reactable';
import { Row, Col } from 'react-bootstrap';
import Breadcrumb from '../../components/layout/Breadcrumb';
import { I18n } from 'react-i18nify';


export const EventsComponent = ({ eventsStore: { object, data, currentLocation } }) => (
  <div>
    <h1>Hendelser</h1>
    <hr />
    <Row>
      <Col sm={2}>
        <b>Museumsnr:</b>
      </Col>
      <Col sm={10}>
        {object && object.museumNo}
      </Col>
    </Row>
    <Row>
      <Col sm={2}>
        <b>Unr:</b>
      </Col>
      <Col sm={10}>
        {object && object.subNo}
      </Col>
    </Row>
    <Row>
      <Col sm={2}>
        <b>Term/Artsnavn:</b>
      </Col>
      <Col sm={10}>
        {object && object.term}
      </Col>
    </Row>
    <Row>
      <Col sm={2}>
        <b>Plassering:</b>
      </Col>
      <Col sm={10}>
        {currentLocation && <Breadcrumb node={currentLocation.breadcrumb} disabled />}
      </Col>
    </Row>
    <br />
    {data &&
      <Table
        className="table"
        columns={[
          { key: 'eventDate', label: 'Dato'},
          { key: 'type', label: 'Type hendelse'},
          { key: 'registeredBy', label: 'Utført av' },
          { key: 'note', label: 'Kommentar' },
          { key: 'id', label: '' }
        ]}
        sortable={['type', 'eventDate', 'registeredBy', 'note' ]}
        defaultSort={{ column: 'eventDate', direction: 'desc' }}
        noDataText={I18n.t('musit.events.noDataForObject')}
      >
        {data.map((event, i) =>
          <Tr key={i}>
            <Td column="eventDate">{event.eventDate}</Td>
            <Td column="type">{event.type}</Td>
            <Td column="registeredBy">{event.registeredBy}</Td>
            <Td column="note"><span>{event.note}</span></Td>
            <Td column="id">Les mer</Td>
          </Tr>
        )}
      </Table>
    }
  </div>
);

export default EventsComponent;