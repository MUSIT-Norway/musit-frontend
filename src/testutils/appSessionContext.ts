// @flow

import * as PropTypes from 'prop-types';
import { Observable } from 'rxjs';

import { AppSession } from '../types/appSession';

export const createAppSessionContext = (appSession: AppSession) => ({
  context: {
    appSession$: Observable.of(appSession)
  },
  childContextTypes: { appSession$: PropTypes.instanceOf(Observable).isRequired }
});

export const createEnLangAppSessionContext = () =>
  createAppSessionContext({
    museumId: 99,
    collectionId: '0000-0000-0000',
    accessToken: 'dummy-token',
    actor: { fn: 'Test Testersen' },
    language: { isEn: false, isNo: true },
    rolesForModules: {
      collectionManagementRead: true,
      collectionManagementWrite: true,
      storageFacilityRead: true,
      storageFacilityWrite: true,
      storageFacilityAdmin: true,
      documentArchiveRead: true,
      documentArchiveWrite: true
    }
  });

export const createNoLangAppSessionContect = () =>
  createAppSessionContext({
    museumId: 99,
    collectionId: '0000-0000-0000',
    accessToken: 'dummy-token',
    actor: { fn: 'Test Testersen' },
    language: { isEn: false, isNo: true },
    rolesForModules: {
      collectionManagementRead: true,
      collectionManagementWrite: true,
      storageFacilityRead: true,
      storageFacilityWrite: true,
      storageFacilityAdmin: true,
      documentArchiveRead: true,
      documentArchiveWrite: true
    }
  });
