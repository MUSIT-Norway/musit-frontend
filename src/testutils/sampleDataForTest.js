// @flow
import type { AppSession } from '../types/appSession';
import type { History } from '../types/Routes';
import type { AnalysisCollection, AnalysisEvent } from '../types/analysis';
import type { SampleDataExtended } from '../types/samples';
import type { MusitObject } from '../types/object';
import type { Field } from '../forms/form';
import { noMapper } from '../forms/mappers';
import {
  getStrField,
  getNumberField,
  getBoolField,
  getArrField,
  getCompositeField
} from '../forms/form';
import type { FormData as AnalysisFormData } from '../modules/analysis/shared/formType';

export const appSession: AppSession = {
  museumId: 99,
  collectionId: '1234',
  accessToken: '45667',
  actor: {
    fn: 'Test'
  },
  language: {
    isEn: false,
    isNo: true
  }
};

export const history: History = {
  push: () => {},
  replace: () => {},
  goBack: () => {},
  goForward: () => {}
};

export const analysis: AnalysisCollection = {
  analysisTypeId: 1,
  eventDate: '2017-03-16T14:37:45+00:00',
  id: 2,
  museumNo: 'MusK58',
  note: 'fdsfsd sdsa 2',
  objectId: 'adea8141-8099-4f67-bff9-ea5090e18335',
  partOf: 1,
  registeredBy: '7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e',
  registeredByName: 'Rituvesh Kumar',
  registeredDate: '2017-04-03T10:36:34+00:00',
  subNo: '2',
  events: [],
  term: 'Mansjettknapp',
  type: 'Analysis'
};

export const sample: SampleDataExtended = {
  id: '123',
  uuid: '0000-0000-123',
  objectId: '0000-0000-123',
  originatedObjectUuid: '0000-0000-123',
  museumId: 99,
  status: 1,
  leftoverSample: 1,
  isExtracted: false,
  isDeleted: false,
  sampleTypeId: 1,
  museumNo: 'M1234',
  term: 'Carex saxatilis',
  subNo: 'a',
  objectType: 'collection',
  currentLocation: { pathNames: null, breadcrumb: null },
  parentObject: {
    objectId: '000-0000-0001',
    objectType: 'collection',
    sampleOrObjectData: {}
  },
  registeredDate: 'some date',
  registeredStamp: { user: '000-0000-0001', date: '2017' },
  // updatedStamp: {name: null},
  objectUUID: '0000-0000-123',
  sampleSubType: 'some sub type',
  doneBy: 'none',
  hasAnalyse: false,
  date: 'some date',
  breadcrumb: [],
  details: 'some details',
  nodeId: 'blee',
  sampleNum: 12,
  sampleType: {
    sampleTypeId: 1,
    sampleType: null,
    enSampleType: 'en',
    noSampleType: 'no',
    noSampleSubType: null,
    enSampleSubType: null
  }
};

export function createAnalysisEventWithObject(
  id: number,
  uuid: string,
  museumNo: string,
  subNo: string,
  term: string
): AnalysisEvent {
  return {
    id: 2 * id,
    affectedThing: uuid,
    affectedType: 'collection',
    analysisTypeId: 5,
    doneBy: 'some uuid',
    doneDate: 'some date',
    partOf: 3 * id,
    registeredBy: 'some uuid',
    registeredDate: 'some date',
    responsible: 'some uuid',
    type: 'Analysis',
    objectData: createObject(id, uuid, museumNo, subNo, term)
  };
}

export function createObject(
  id: number,
  uuid: string,
  museumNo: string,
  subNo: string,
  term: string
): MusitObject {
  return {
    id: id,
    uuid: uuid,
    museumId: 99,
    museumNo: museumNo,
    subNo: subNo,
    term: term,
    collection: 8,
    materials: [],
    locations: [],
    coordinates: [],
    objectType: 'collection'
  };
}

export const object: MusitObject = {
  id: 56,
  uuid: '12080e3e-2ca2-41b1-9d4a-4d72e292dcd8',
  museumId: 99,
  museumNo: 'MusN11',
  term: 'Solsikke',
  collection: 8,
  materials: [],
  locations: [],
  coordinates: [],
  objectType: 'collection'
};

export function createField<T>(
  name: string,
  value: T,
  valid?: boolean = false
): Field<T> {
  return {
    name: name,
    rawValue: noMapper.toRaw(value),
    defaultValue: value,
    status: {
      valid: valid
    },
    validator: {},
    mapper: noMapper
  };
}

export const analysisForm: AnalysisFormData = {
  id: getNumberField('', 1),
  analysisTypeId: getNumberField('', 1),
  analysisTypeCategory: getStrField('', '1'),
  doneBy: getStrField('', '1'),
  doneDate: getStrField('', '1'),
  registeredBy: getStrField('', '1'),
  registeredByName: getStrField('', '1'),
  registeredDate: getStrField('', '1'),
  responsible: getStrField('', '1'),
  administrator: getStrField('', '1'),
  persons: getArrField('', [
    {
      name: 'jarl',
      role: 'responsible'
    },
    {
      name: 'line',
      role: 'creator',
      date: '2017-06-29T07:54:22+00:00'
    }
  ]),
  completedBy: getStrField('', '1'),
  completedDate: getStrField('', '1'),
  objectId: getStrField('', '1'),
  note: getStrField('', '1'),
  type: getStrField('', '1'),
  partOf: getStrField('', '1'),
  result: getStrField('', '1'),
  orgId: getNumberField('orgId', 1),
  externalSource: getStrField('', '1'),
  comments: getStrField('', '1'),
  reason: getStrField('', '1'),
  caseNumbers: getArrField('', []),
  completeAnalysis: getStrField('', '1'),
  // restriction
  restrictions: getBoolField('', true),
  restriction: getCompositeField('', {
    requester: '3cbf15cb-8348-4e66-99a4-bc314da57444',
    requesterName: 'Jarl',
    reason: 'fin Ã¥rsak',
    caseNumbers: ['dddd', '44555', '55555'],
    expirationDate: '2017-03-29T07:54:22+00:00',
    cancelledReason: 'meh'
  }),
  // object details
  museumNo: getStrField('', '1'),
  subNo: getStrField('', '1'),
  term: getStrField('', '1'),
  eventDate: getStrField('', '1'),
  actor: getStrField('', '1'),
  events: getArrField('', []),
  updatedBy: getStrField('', '1'),
  updatedByName: getStrField('', '1'),
  updatedDate: getStrField('', '1'),
  status: getNumberField('status', 1)
};
