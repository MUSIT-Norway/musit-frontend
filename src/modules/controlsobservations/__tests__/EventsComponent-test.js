import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { EventsComponent } from '../EventsComponent';
import { appSession } from '../../../testutils/sampleDataForTest';

describe('EventsComponent', () => {
  const history = {
    push: () => {}
  };

  it('Check it renders when not loaded', () => {
    const myDiv = shallow(
      <EventsComponent
        store={{
          data: [],
          loading: true
        }}
        loader={<span>loader</span>}
        showObservations={true}
        showControls={true}
        match={{
          params: {
            id: '1'
          }
        }}
        appSession={appSession}
        clearEvents={() => true}
        loadEvents={() => true}
        loadRootNode={() => true}
        history={history}
      />
    );
    expect(shallowToJson(myDiv)).toMatchSnapshot();
  });

  it('Check it renders.', () => {
    const myDiv = shallow(
      <EventsComponent
        store={{
          data: []
        }}
        showObservations={true}
        showControls={true}
        match={{
          params: {
            id: '1'
          }
        }}
        appSession={appSession}
        clearEvents={() => true}
        loadEvents={() => true}
        loadRootNode={() => true}
        history={history}
      />
    );
    expect(shallowToJson(myDiv)).toMatchSnapshot();
  });
});
