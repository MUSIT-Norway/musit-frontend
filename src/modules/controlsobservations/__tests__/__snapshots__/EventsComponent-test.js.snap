// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsComponent Check it renders when not loaded 1`] = `
<Layout
  breadcrumb={
    <Breadcrumb
      allActive={true}
      node={undefined}
      onClickCrumb={[Function]}
    />
  }
  content={
    <span>
      loader
    </span>
  }
  leftMenu={
    <div
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <ObservationControlComponent
        id="1"
        onClickNewControl={[Function]}
        onClickNewObservation={[Function]}
      />
    </div>
  }
  title="Magasin"
  toolbar={
    <Toolbar
      clickShowCenter={[Function]}
      clickShowLeft={[Function]}
      hideRight={true}
      labelCenter="Kontroller"
      labelLeft="Observasjoner"
      placeHolderSearch="Filtrer i listen"
      showCenter={true}
      showLeft={true}
    />
  }
/>
`;

exports[`EventsComponent Check it renders. 1`] = `
<Layout
  breadcrumb={
    <Breadcrumb
      allActive={true}
      node={undefined}
      onClickCrumb={[Function]}
    />
  }
  content={
    <div
      style={
        Object {
          "color": "grey",
          "textAlign": "center",
        }
      }
    >
      Ingen kontroller eller observasjoner
    </div>
  }
  leftMenu={
    <div
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <ObservationControlComponent
        id="1"
        onClickNewControl={[Function]}
        onClickNewObservation={[Function]}
      />
    </div>
  }
  title="Magasin"
  toolbar={
    <Toolbar
      clickShowCenter={[Function]}
      clickShowLeft={[Function]}
      hideRight={true}
      labelCenter="Kontroller"
      labelLeft="Observasjoner"
      placeHolderSearch="Filtrer i listen"
      showCenter={true}
      showLeft={true}
    />
  }
/>
`;
